be/src/olap/tablet_meta.cpp

```diff
@@ -17,968 +17,197 @@
-using std::ifstream;
-using std::ios;
-using std::list;
-using std::make_pair;
-using std::ofstream;
-using std::queue;
-using std::sort;
-using std::string;
-using std::stringstream;
-using std::unordered_map;
-using std::vector;
-
 namespace doris {
-// related static functions of version graph
-
-// Construct version graph(using adjacency list) from header's information.
-static OLAPStatus construct_version_graph(
-        const RepeatedPtrField<PDelta>& versions_in_header,
-        vector<Vertex>* version_graph,
-        unordered_map<int, int>* vertex_helper_map);
-
-// Clear version graph and vertex_helper_map, release memory hold by version_graph.
-static OLAPStatus clear_version_graph(vector<Vertex>* version_graph,
-                                  unordered_map<int, int>* vertex_helper_map);
-
-// Add version to graph, it is called near the end of add_version
-static OLAPStatus add_version_to_graph(const Version& version,
-                                   vector<Vertex>* version_graph,
-                                   unordered_map<int, int>* vertex_helper_map);
-
-// Delete version from graph, it is called near the end of delete_version
-static OLAPStatus delete_version_from_graph(
-        const RepeatedPtrField<PDelta>& versions_in_header,
-        const Version& version,
-        vector<Vertex>* version_graph,
-        unordered_map<int, int>* vertex_helper_map);
 
-// Add vertex to graph, if vertex already exists, still return SUCCESS.
-static OLAPStatus add_vertex_to_graph(int vertex_value,
-                                  vector<Vertex>* version_graph,
-                                  unordered_map<int, int>* vertex_helper_map);
-
-TabletMeta::~TabletMeta() {
-    // Release memory of version graph.
-    clear_version_graph(&_version_graph, &_vertex_helper_map);
-    Clear();
+TabletMeta::TabletMeta(DataDir* data_dir) {
+    _data_dir = data_dir;
 }
 
-OLAPStatus TabletMeta::init() {
-    clear_version_graph(&_version_graph, &_vertex_helper_map);
-    if (construct_version_graph(delta(),
-                                &_version_graph,
-                                &_vertex_helper_map) != OLAP_SUCCESS) {
-        OLAP_LOG_WARNING("fail to construct version graph.");
-        return OLAP_ERR_OTHER_ERROR;
-    }
-    if (_file_name == "") {
-        stringstream file_stream;
-        file_stream << tablet_id() << ".hdr";
-        _file_name = file_stream.str();
-    }
-    return OLAP_SUCCESS;
-}
+NewStatus TabletMeta::serialize(string* meta_binary) {
+    std::lock_guard<std::mutex> lock(_mutex);
+    return serialize_unlock(meta_binary);
+};
 
-OLAPStatus TabletMeta::load_and_init() {
-    // check the tablet_path is not empty
-    if (_file_name == "") {
-        LOG(WARNING) << "file_path is empty for header";
-        return OLAP_ERR_DIR_NOT_EXIST;
-    }
-
-    FileHeader<OLAPHeaderMessage> file_header;
-    FileHandler file_handler;
+NewStatus TabletMeta::serialize_unlock(string* meta_binary) {
```

Comment:
it is better to us unprotect_serialize()

Reply:
Rocksdb and kudu all use unlock, so I adopt this name.